addons/stock/models/stock_picking.py

```diff
@@ -100,7 +100,10 @@ def _default_show_operations(self):
     def create(self, vals_list):
         for vals in vals_list:
             if 'sequence_id' not in vals or not vals['sequence_id']:
-                if vals['warehouse_id']:
+                if not self.user_has_groups('stock.group_stock_multi_warehouses') and not vals.get('warehouse_id'):
```

Comment:
I would transform warehouse_id as a compute store true. That depends on default_location_src_id and default_location_dest_id. The current solution is a bit hacky